systest/tests/smeshing_test.go

```diff
@@ -142,3 +164,221 @@ func requireEqualEligibilities(tctx *testcontext.Context, tb testing.TB, proposa
 		}
 	}
 }
+
+func testVesting(tb testing.TB, tctx *testcontext.Context, cl *cluster.Cluster, accs ...vestingAcc) {
+	tb.Helper()
+	var (
+		eg      errgroup.Group
+		genesis = cl.GenesisID()
+	)
+	for i := range accs {
+		acc := accs[i]
+		client := cl.Client(i % cl.Total())
+		eg.Go(func() error {
+			var subeg errgroup.Group
+			watchLayers(tctx, &subeg, client, func(layer *pb.LayerStreamResponse) (bool, error) {
+				return layer.Layer.Number.Number < uint32(acc.start), nil
+			})
+			if err := subeg.Wait(); err != nil {
+				return err
+			}
+
+			ctx, cancel := context.WithTimeout(tctx, 10*time.Minute)
+			defer cancel()
+			var nonce uint64
+			id, err := submitTransaction(ctx, acc.selfSpawn(genesis, nonce), client)
+			if err != nil {
+				return fmt.Errorf("selfspawn multisig: %w", err)
+			}
+			nonce++
+
+			initial, err := currentBalance(ctx, client, acc.address)
+			if err != nil {
+				return err
+			}
+			tctx.Log.Debugw("submitted selfspawn",
+				"address", acc.address,
+				"initial", initial,
+			)
+			waitTransaction(ctx, &subeg, client, id)
+			if err := subeg.Wait(); err != nil {
+				return err
+			}
+			id, err = submitTransaction(ctx, acc.spawnVault(genesis, nonce), client)
+			if err != nil {
+				return fmt.Errorf("spawn vault: %w", err)
+			}
+			nonce++
+			tctx.Log.Debugw("submitted spawn vault",
+				"address", acc.vault,
+			)
+			waitTransaction(ctx, &subeg, client, id)
+			if err := subeg.Wait(); err != nil {
+				return err
+			}
+			leftover := acc.total
+			for leftover > 0 {
+				step := acc.total / (acc.end - acc.start)
+				if leftover > step {
+					leftover -= step
+				} else {
+					step = leftover
+					leftover = 0
+				}
+				tctx.Log.Debugw("submitted drain vault",
+					"amount", step,
+					"leftover", leftover,
+				)
+				id, err := submitTransaction(ctx, acc.drain(genesis, uint64(step), nonce), client)
+				if err != nil {
+					return fmt.Errorf("drain: %w", err)
+				}
+				nonce++
+				waitTransaction(ctx, &subeg, client, id)
+				if err := subeg.Wait(); err != nil {
+					return err
+				}
+			}
+			remaining, err := currentBalance(ctx, client, acc.vault)
+			if err != nil {
+				return err
+			}
+			current, err := currentBalance(ctx, client, acc.address)
+			if err != nil {
+				return err
+			}
+			tctx.Log.Infow("results for account after tests",
+				"vest", acc.address,
+				"vault", acc.vault,
+				"initial", initial,
+				"current", current,
+				"vested", acc.total,
+				"remaining", remaining,
+				"client", client.Name,
+			)
+			if remaining != 0 {
+				return fmt.Errorf("vault at %v must be empty, instead has %d", acc.vault, remaining)
+			}
+			if delta := int(current - initial); delta+1e7 < acc.total {
```

Comment:
- can you show how 1e7 is caclucated
- why isn't this `delta+1e7 != acc.total`?

Reply:
1e7 is just a small number that should cover all fees. it is not calculated, i just guesstimated it

Reply:
ok. whoever works on this area later will not easily understand this.